### PR TITLE
deps: upgrade to Angular 12 with Webpack 5

### PR DESCRIPTION
**Pull request by dhhyi**
## PR Type

[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no API changes)
[x] Build-related changes

## What Is the Current Behavior?

Angular 11 with Webpack 4

## What Is the New Behavior?

Angular 12 with Webpack 5

## Does this PR Introduce a Breaking Change?

[x] Yes
[ ] No

## Other Information

- bootstrap pinned to version 4
- some jest dependencies (jest@27, jest-preset-angular@9) fail tests on upgrade -- maybe next time 😅 
- prettier upgrade with formatting change as extra commit << please do rebase merge

replaces #699

### CODE REVIEW

**Comment by jometzner** [comment]
Looks good. I wasn't that far away with my custom.webpack.ts

File: angular.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
               "namedChunks": false,
               "extractLicenses": true,
               "vendorChunk": true,
+              "commonChunk": false,
```
Comment (jometzner) [inline angular.json:57]: Why are you disabling the common bundle?

Comment (jometzner) [reply angular.json:57]: Speed up loading time? Only load what is really needed etc.?

Comment (dhhyi) [reply angular.json:57]: it's not disabled, 'technically'. This is just preventing that Angular inserts a default configuration. The webpack.custom still pushes artifacts into the common bundle and it is being created.

Comment (jometzner) [reply angular.json:57]: True that. I was wondering why `ng build --stats-json` still spits out common.js.

**Review by jometzner — APPROVED** [review]

**Pull request merged**